http/handler_extras_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package http
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/sourcenetwork/defradb/event"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPurge(t *testing.T) {
```

Comment:
todo: Please also add integration tests that show that purging when in production mode will not work and also that purging while in development mode will in fact purge the database.

Reply:
Discussed on Discord. The `start` command cannot currently be tested with the integration test suite and `purge` is handled within the `start` command. We will find a different way to test the effectiveness of the `purge` command.

Reply:
> The start command cannot currently be tested with the integration test suite and purge is handled within the start command

I'm curious - why can it not be tested with integration tests?

Reply:
> > The start command cannot currently be tested with the integration test suite and purge is handled within the start command
> 
> I'm curious - why can it not be tested with integration tests?

It's possible but there are a few things that make it really difficult and not worth the effort in my opinion. No easy way for random port assignments and the timing of the command (waiting for start up / restart) will cause the tests to be flaky.

Reply:
> Why would any of these tests need to care about port assignment?

To test the purge command you need the http api running.

Reply:
Oh you changed it to go via the http client?  Is that not exposing a highly destructive command a little too much? I wonder if that was an oversight in our last conversation.

Reply:
> Oh you changed it to go via the http client? Is that not exposing a highly destructive command a little too much? I wonder if that was an oversight in our last conversation.

That's why I've added the development mode config value that must be enabled to run purge.

Reply:
Yeah, I assumed it would still only be available locally though, not over http too.

People will still mess up and accidentally set the dev mode to on in production deployments, and at the moment (prior to admin ACP, and correct config of it...) it would allow world plus dog to truncate the db.

Reply:
We can't put guardrails against devs that shoot themselves in the foot to that extent. Devs can also just expose a production DB without any ACP policies. It would be just as bad. At least here we have to be explicit with putting the database in dev mode to allow the purge command.